src/PrimitiveObsession/IoC/EngineContainerBuilder.cs

```diff
@@ -17,8 +17,7 @@ public IContainer Build()
         {
             var builder = new ContainerBuilder();
 
-            builder.RegisterModule(new ConfigModule(_config));
-            builder.RegisterModule(new MainModule());
+            builder.RegisterModule(new PrimitiveObsessionModule(_config));
```

Comment:
Guarda come AutoFac definisce i moduli: dice che sono

"a small class that can be used to bundle up a
set of related components behind a ‘facade’ to simplify configuration and
deployment."

Io sinceramente non vedo l'utilità in questo caso di avere 2 moduli: tutte le classi coinvolte sono già "bundled" insieme. Non c'è davvero motivo di gestirle in due gruppi diversi. A me sembra una complicazione che possiamo semplificare.

Il senso è sempre quello: secondo me per il codice dell'articolo dobbiamo cercare di eliminare ogni forma di complessità che possa rendere più faticolo al lettore l'interpretazione del nostro esempio.

In realtà, sono sincero, unirei in un solo modulo in tutti i casi, anche se fosse codice di produzione.